### DISCUSSION TITLE
Using iperf3 to test jitter

### DISCUSSION THREAD

**Discussion by fatman00** [esnet/iperf#1581] [Q&A] [answered]
Hi there,

Is there anyway I can get iperf3 to make a long running UDP connection where it will send small UDP packets but often.
I would like to send 1mbit/s packets but send a packet every 2 ms. Or I want to send 100bytes backets every 5 ms.

**Comment by davidBar-On** [comment, marked as answer]
Is it important that the packets will be sent exactly every 2ms or 5ms? If not, then iperf3 tries to split the packets evenly over time, but the delay between the packets is not exactly the average delay.  To send 100 bytes packets every 5 ms, i.e. 200 packets/sec, you can therefore try `iperf3 -c <host> -l 100 -b 160000` (160,000 = 100(bytes) * 200 * 8(bits/byte)).  Similarly, for 1Mbits/sec every 2ms, i.e. 500 packets/sec, try `iperf3 -c <host> -l 262 -b 1M` (262 is about "1M / 8 / 500").

If this will not work well enough for you, there is an iperf3 version that was not merged into the mainline that may help (assuming you can build iperf3 yourself): #1343.  See the PR description for details.

**Reply by fatman00** [reply]
Hi @davidBar-On,
Thank you for the examples I will try them out.
It is not that important that they are send every 2/5 ms as long as they are sent with the same interval, as I want to make some statistics about the buffer delay on the traffic path.
